backend/backend/MathGame/MathGameController.cs

```diff
@@ -78,21 +82,32 @@ private static bool CheckPuzzleAnswer(string puzzle, string answer)
         try
         {
             var tokens = puzzle.Split(' ');
-            int leftOperand = int.Parse(tokens[0]);
-            string operation = tokens[1];
-            int rightOperand = int.Parse(tokens[2]);
+            if (tokens.Length < 3)
+            {
+                return false;
+            }
+
+            if (int.TryParse(tokens[0], NumberStyles.Integer, CultureInfo.InvariantCulture, out int leftOperand) &&
+                int.TryParse(tokens[2], NumberStyles.Integer, CultureInfo.InvariantCulture, out int rightOperand))
+            {
+                string operation = tokens[1];
+
+                int correctAnswer = operation switch
+                {
+                    "+" => leftOperand + rightOperand,
+                    "-" => leftOperand - rightOperand,
+                    "*" => leftOperand * rightOperand,
+                    "/" => rightOperand != 0 ? leftOperand / rightOperand : 0,
+                    _ => 0
+                };
 
-            // Solve the puzzle based on the operator
-            int correctAnswer = operation switch
+                return int.TryParse(answer, NumberStyles.Integer, CultureInfo.InvariantCulture, out int userAnswer) &&
+                       userAnswer == correctAnswer;
+            }
+            else
             {
-                "+" => leftOperand + rightOperand,
-                "-" => leftOperand - rightOperand,
-                "*" => leftOperand * rightOperand,
-                "/" => leftOperand / rightOperand,
-                _ => 0
-            };
-
-            return int.Parse(answer) == correctAnswer;
+                return false; // Invalid puzzle format or parsing issue
+            }
```

Comment:
Siaip not a big deal, bet geriau butu early returnas. Paprasciau skaityt, daugiau laisvos horizontal space ir nereiks komentaro prie `return false;`.
```
if (!valid_puzzle) {
    return false;
}

string operation = tokens[1];
...
```